### ISSUE TITLE
Updating Readme does not work any more

### ISSUE THREAD

**Issue by ugroempi** [bertcarnell/SOAs#29]
Hi Rob,
I don't understand the error message - apparently the file is rendered, but then something goes wrong (something about an untracked file that I can't locate in folder .github).
Would you know how to fix this?
Best, Ulrike

**Comment by bertcarnell** [comment]
I will take a look

**Comment by bertcarnell** [comment]
I updated the github actions to hopefully resolve this issue.  I won't close the issue until we see if it is actually fixed.  Here is what I intend to happen:

- **r-cmd-check.yml**:  Run on changes to almost everything except:
    - tags
    - changes to ancillary files
    - pushes to non-main branches
- **build_readme.yaml**:  Run only on changes to files that affect it
    - README.md and man/SOAs-package.Rd in the "main"branch
    - don't run on tags anymore
- **pkgdown.yml**:  run only on a tag

The error you were seeing in the actions output was that the process of checking the README.md back into the repository doesn't work if the tag was checked out (you can't push back to a tag).  So, I'll just prevent it from running on a tag.

**Comment by ugroempi** [comment]
Thanks! This works now - the layout problem is still there, but that's a different story, and I can now try to fix it.

**Comment by ugroempi** [comment]
Hmmh, worked just once, but now the second attempt failed again ... ???

**Comment by bertcarnell** [comment]
This time the failure was due to no changes being detected.  When you made your change to Readme.Rmd and Readme.md was created, the **new** Readme.md was exactly the same as the **old** Readme.md so git had nothing to check in.

```
Run git config --local user.name "$GITHUB_ACTOR"
  git config --local user.name "$GITHUB_ACTOR"
  git config --local user.email "$GITHUB_ACTOR@users.noreply.github.com"
  git add README.md
  git commit -m 'Automatic build of README.md'
  git push
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_PAT: ***
    R_LIBS_USER: /home/runner/work/_temp/Library
    TZ: UTC
    _R_CHECK_SYSTEM_CLOCK_: FALSE
    NOT_CRAN: true
    RSPM: https://packagemanager.rstudio.com/all/__linux__/focal/latest
On branch main
Your branch is up to date with 'origin/main'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.github/r-depends.rds

nothing added to commit but untracked files present (use "git add" to track)
Error: Process completed with exit code 1.
```

The clue is that `your branch is up to date with 'origin/main'` and `nothing added to commit` which means that we tried to add Readme.md to the files for commit, but it was not changed.

When it does work, we get something like this ...

```
Run git config --local user.name "$GITHUB_ACTOR"
  git config --local user.name "$GITHUB_ACTOR"
  git config --local user.email "$GITHUB_ACTOR@users.noreply.github.com"
  git add README.md
  git commit -m 'Automatic build of README.md'
  git push
  shell: /usr/bin/bash -e {0}
  env:
    GITHUB_PAT: ***
    R_LIBS_USER: /home/runner/work/_temp/Library
    TZ: UTC
    _R_CHECK_SYSTEM_CLOCK_: FALSE
    NOT_CRAN: true
    RSPM: https://packagemanager.rstudio.com/all/__linux__/focal/latest
[main b563140] Automatic build of README.md
 1 file changed, 10 insertions(+), 5 deletions(-)
To https://github.com/bertcarnell/SOAs
   46f8[2](https://github.com/bertcarnell/SOAs/runs/7713658769?check_suite_focus=true#step:7:2)48..b56[3](https://github.com/bertcarnell/SOAs/runs/7713658769?check_suite_focus=true#step:7:3)1[4](https://github.com/bertcarnell/SOAs/runs/7713658769?check_suite_focus=true#step:7:4)0  main -> main
```

**Comment by bertcarnell** [comment]
I think I can add a test to the script to see if there are changes and give us a more useful error.

**Comment by bertcarnell** [comment]
What I suspected seems to be correct.  There was no difference between the newly created README.md  and the one already in the repo.  The changes you made didn't result in a visible change after rendering.

**Comment by ugroempi** [comment]
Thanks for figuring out the problem - maybe this was the problem all along, and it was not related to tags? But anyway, I like the new way of triggering the Readme on changes to `SOAs-package.Rd`. 

Off-topic: I just have to figure out why the equation `\eqn{D = s^2 A + s B + C}` is messed up in the Readme, it looks OK in the documentation. I have no clue what could be wrong with it, which is why I tried a line break that apparently did not make a difference. Would you happen to have an idea?

**Comment by ugroempi** [comment]
I have moved the off-topic Readme problem to an issue of its own. 
A more informative error message for failures because of no change in the md would be good, if possible.

**Comment by bertcarnell** [comment]
I think all the elements of this thread are resolved with the latest commit 80332c84d2ab7de8ed68495bf9cb16e8b034d5dc
